lib/jekyll/excerpt.rb

```diff
@@ -116,11 +116,28 @@ def place_in_layout?
     def extract_excerpt(doc_content)
       head, _, tail = doc_content.to_s.partition(doc.excerpt_separator)
 
+      # append appropriate closing tag (to a Liquid block), to the "head" if the
+      # partitioning resulted in leaving the closing tag somewhere in the "tail"
+      # partition.
+      if head =~ %r!{%\s*(\w+).+\s*%}!
```

Comment:
Regexp's are _very_ slow! Can we use `head.include?("{%")` before we do regexp? I think this runs on every page generation...